### ISSUE TITLE
Editor crashed after entering an incorrect HTML tag in the source editing mode

### ISSUE THREAD

**Issue by psmyrek** [ckeditor/ckeditor5#12112]
## 📝 Provide detailed reproduction steps (if any)

1. Build the docs (`yarn run docs`) and start the docs server (`yarn run docs:serve`).
2. Open source editing feature guide: http://127.0.0.1:8080/ckeditor5/latest/features/source-editing.html#demo.
3. In the source editing demo, remove the existing content and switch to the **source mode**.
4. Enter some invalid HTML tag, e.g. `<p`.
5. Exit the **source mode**.

### ✔️ Expected result

Editor should not crash.

### ❌ Actual result

Editor crashed:

```
Uncaught Error: Failed to execute 'createElement' on 'Document': The tag name provided ('p<') is not a valid name.
    at DomConverter.viewToDom (domconverter.js:419:31)
    at DomConverter.viewChildrenToDom (domconverter.js:535:16)
    at viewChildrenToDom.next (<anonymous>)
    at DomConverter.viewToDom (domconverter.js:439:17)
    at HtmlDataProcessor.toData (htmldataprocessor.js:65:41)
    at model (customelement.js:106:48)
    at getModelElement (upcasthelpers.js:815:10)
    at UpcastDispatcher.eval (upcasthelpers.js:791:24)
    at UpcastDispatcher.fire (emittermixin.ts:183:43)
    at UpcastDispatcher._convertItem (upcastdispatcher.js:270:9)
```

## 📃 Other details

* Browser: Firefox 102 and Chrome 103
* OS: Windows 10
* First affected CKEditor version: **34.2.0**

This test scenario does not break the editor in version **34.1.0** and earlier.

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by niegowski** [comment]
This looks like broken parsing of HTML string:
![image](https://user-images.githubusercontent.com/1232187/180393684-006970fb-6141-48a5-9562-d0238af0308d.png)

**Comment by Reinmar** [comment]
Options:

*   Check the element name inside the upcast converter that tries to pick it up: [https://github.com/ckeditor/ckeditor5/blob/ef39bdfb53aedfbe0bfd1ac2b5ed384e22aad46c/packages/ckeditor5-html-support/src/integrations/customelement.js#L62-L73.](https://github.com/ckeditor/ckeditor5/blob/ef39bdfb53aedfbe0bfd1ac2b5ed384e22aad46c/packages/ckeditor5-html-support/src/integrations/customelement.js#L62-L73.) Let's have allowlist-regexp filtering out elements that have incorrect names. We can try to check out the spec for `document.createElement()` what kind of element names are allowed.
*   Verify the element name inside `viewToDom()` because that's where the crash happens. Or just use try-catch. However, this piece of code is really hot so it'd be good to avoid increasing its complexity.

**Comment by Reinmar** [comment]
BTW, this is fun :facepalm: 

![](https://user-images.githubusercontent.com/156149/180420194-e8360551-45e8-4716-9f25-67299b6fe95d.png)

**Comment by Reinmar** [comment]
BTW, the above means that another piece of HTML that'll break this is `<p` :

![](https://user-images.githubusercontent.com/156149/180420489-6c058073-a7a1-4737-9f40-273597458085.png)

**Comment by Reinmar** [comment]
:point\_up: The above issue is not anymore related to custom elements (the first one was and the fix could be located in the custom elements handler). This is the "Allow more attributes on an existing element" feature of GHS.

**Comment by Reinmar** [comment]
Scope:

*   Verify element names inside the aforementioned upcast converter for custom elements.
*   Verify (filter out) incorrect attribute names in `dataFilter.processViewAttributes()`.
*   Adding tests for:
    *   the two scenarios mentioned in this ticket
    *   since `processViewAttributes()` is used in a couple more GHS's integrations (e.g. for images, media embed) and cases like support for `div`s, trying to come up with 2-3 more scenarios (basically - for different element names than `p`)

**Comment by Reinmar** [comment]
Note: Be careful to not break support for e.g. namespaced elements and custom elements like (`<foo-bar>`).

**Comment by niegowski** [comment]
Closed in #12175.
